tools/merge_archives.ml

```diff
@@ -0,0 +1,106 @@
+(**************************************************************************
+ *                                                                        *
+ *                                 OCaml                                  *
+ *                                                                        *
+ *                    Mark Shinwell, Jane Street Europe                   *
+ *                                                                        *
+ *   Copyright 2021 Jane Street Group LLC                                 *
+ *                                                                        *
+ *   All rights reserved.  This file is distributed under the terms of    *
+ *   the GNU Lesser General Public License version 2.1, with the          *
+ *   special exception on linking described in the file LICENSE.          *
+ *                                                                        *
+ **************************************************************************)
+
+module String_set = Set.Make (String)
+
+let merge_cma ~target:_ ~archives:_ =
+  failwith "not yet implemented"
+
+let read_cmxa filename =
+  let chan = open_in_bin filename in
+  let magic =
+    really_input_string chan (String.length Config.cmxa_magic_number)
+  in
+  let (cmxa : Cmx_format.library_infos) = input_value chan in
```

Comment:
It looks like we should check the magic number here;
there is a consistency check in the function below, but
my fear is the unmarshaling of something with the wrong
shape (and the segfault that could follow).

Or is it written this way because you would like to be
compatible across versions?

Reply:
It deliberately doesn't check the magic number so it can be used across versions (it used to, and was problematic).

Reply:
I think it has the potential of failing badly, but again this is for
power users so we can let them shoot themselves in the foot
since they are powerful enough to recover.

Reply:
Ah, I see, because you mean that "library_infos" might have changed.  I think this is probably ok for an internal build system tool.

Reply:
Yes, that type or any one transitively used from it.
In the end I agree it is fine; the tool is simple enough so that
if we see an unexplained failure, it will come from the unmarshaling.